node/src/Unison/Runtime/Cryptography.hs

```diff
@@ -0,0 +1,85 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# Language ScopedTypeVariables #-}
+
+module Unison.Runtime.Cryptography where
+
+import Unison.Cryptography
+import Data.ByteString (ByteString)
+import Data.ByteArray as BA
+import qualified Crypto.Random as R
+
+-- cryptonite
+-- import Crypto.Cipher.Types.Base ()
+-- import Crypto.Cipher.AES.Primitive (AES)
+import qualified Crypto.Cipher.AES as AES
+import Crypto.Cipher.Types
+import Crypto.Error
+
+-- Creates a Unison.Cryptography object specialized to use the noise protocol
+-- (http://noiseprotocol.org/noise.html).
+mkCrypto :: forall cleartext symmetricKey . (ByteArrayAccess cleartext, ByteArray cleartext, ByteArrayAccess symmetricKey, ByteArray symmetricKey) => ByteString -> Cryptography ByteString symmetricKey () () () () cleartext
+mkCrypto key = Cryptography key gen hash sign verify randomBytes encryptAsymmetric decryptAsymmetric encrypt decrypt pipeInitiator pipeResponder where
+  -- generates an elliptic curve keypair, for use in ECDSA
+  gen = undefined
+  hash = undefined
+  sign _ = undefined
+  verify _ _ _ = undefined
+  randomBytes = randomBytes'
+  encryptAsymmetric _ cleartext = undefined
+  decryptAsymmetric ciphertext = undefined
+
+  encrypt :: symmetricKey -> [cleartext] -> IO Ciphertext
+  encrypt = encrypt'
+      
+  decrypt :: symmetricKey -> ByteString -> Either String cleartext
+  decrypt = decrypt'
+
+  pipeInitiator _ = undefined
+  pipeResponder = undefined
+
+randomBytes' :: Int -> IO ByteString
+randomBytes' n = do drg <- R.getSystemDRG
+                    let bts = fst $ R.randomBytesGenerate n drg
+                    return bts
+
+ivBitLength :: Int
+ivBitLength = 128
+
+authTagBitLength :: Int
+authTagBitLength = 128
+
+encrypt' :: forall cleartext symmetricKey .
+            ( ByteArrayAccess cleartext
+            , ByteArray cleartext
+            , ByteArrayAccess symmetricKey
+            , ByteArray symmetricKey)
+         => symmetricKey
+         -> [cleartext]
+         -> IO Ciphertext
+encrypt' k cts = do
+  iv <- randomBytes' $ ivBitLength `div` 8
```

Comment:
Again, would do this via a `fmap` - `go <$> randomBytes' (ivBitLength `div` 8) where go iv = ...`